internal/publish/metrics.go

```diff
@@ -29,6 +29,9 @@ var (
 
 	mLatencyMs = stats.Float64(publishMetricsPrefix+"requests", "publish requests", stats.UnitMilliseconds)
 
+	mNoPublicKeys = stats.Int64(publishMetricsPrefix+"no_pub_keys",
```

Comment:
Nit, this should be:

```suggestion
	mNoPublicKey = stats.Int64(publishMetricsPrefix+"no_public_key",
```